### ISSUE TITLE
Feat: add more docs?

### ISSUE THREAD

**Issue by ololosha228** [bspaans/bleep#11]
Well, you know, docs are great! Personally i didn't understood, what's differences between generator and channel, why some effects are hardcoded, etc. I can contribute, but i don't understand most of mechanics in this package

**Comment by bspaans** [comment]
Yeah, the project is a bit light on docs. I think what could be useful is to have a short tutorial with code samples to talk a bit more on how to achieve certain things, because I'm not sure godoc is the right place for it (although the godocs certainly could use some love as well)

To answer your question: generators are the basic sound generating building block; usually you want your generator to play certain notes at certain time intervals etc. so in this case you can put your generator in a channel, which has e.g. NoteOn and NoteOff methods to control when the generator outputs samples.  The Sequencer implementation takes multiple of these channels (described as `ChannelDef`s) and is then able to do automations, play MIDI, etc. Hope that helped a bit.
